genie-web/src/main/java/com/netflix/genie/web/controllers/GenieExceptionMapper.java

```diff
@@ -137,4 +141,21 @@ public void handleConstraintViolation(
         log.error(cve.getLocalizedMessage(), cve);
         response.sendError(HttpStatus.PRECONDITION_FAILED.value(), builder.toString());
     }
+
+    /**
+     * Handle MethodArgumentNotValid  exceptions.
+     *
+     * @param response The HTTP response
+     * @param e        The exception to handle
+     * @throws IOException on error in sending error
+     */
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public void handleMethodArgumentNotValidException(
+        final HttpServletResponse response,
+        final MethodArgumentNotValidException e
+    ) throws IOException {
+        this.methodArgumentNotValidRate.increment();
+        log.error(e.getMessage(), e);
+        response.sendError(HttpStatus.PRECONDITION_FAILED.value(), e.getMessage());
```

Comment:
This is a good point. I wasn't aware the spec was just based on Header validation. Originally used 412 because just considered it a precondition failed like a method call internally.

I won't change this as part of this PR or in 3.0.x cause it would change way too much by way of API across the board and for clients. We can explore this in 3.1.x